### ISSUE TITLE
clarify @define_diffrule namespaces

### ISSUE THREAD

**Issue by tpapp** [JuliaDiff/DiffRules.jl#17]
I think that eg
```julia
@define_diffrule SpecialFunctions.lgamma(x) = :( digamma($x) )
```
should be
```julia
@define_diffrule SpecialFunctions.lgamma(x) = :( SpecialFunctions.digamma($x) )
```
since the RHS is not necessarily evaluated in the module of the LHS. Is this correct?

Just please clarify, I would be happy to make a PR.

(cf https://github.com/JuliaStats/StatsFuns.jl/issues/49)

**Comment by jrevels** [comment]
Yup, good catch! The other rules should have this as well.

**Comment by tpapp** [comment]
Can you help me figure out why this wasn't caught by the unit tests? I think they should be updated first, so that we get an error, which is then fixed.

**Comment by tpapp** [comment]
Indeed, `import` breaks the tests, I made a PR which fixes things.

A review would be appreciated, especially for the case I mention in the PR discussion.
